File: app/src/main/java/ekzeget/ru/ekzeget/di/AppComponent.java
package ekzeget.ru.ekzeget.di;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import ekzeget.ru.ekzeget.di.modules.ContextModule;
import ekzeget.ru.ekzeget.di.modules.GithubModule;
import ekzeget.ru.ekzeget.mvp.GithubService;

@Singleton
@Component(modules = {ContextModule.class, GithubModule.class})
public interface AppComponent {
    Context getContext();
    GithubService getAuthService();
}

File: app-kotlin/src/main/java/ru/ekzeget/injection/AppComponent.kt
package ru.ekzeget.injection

import android.content.Context
import dagger.Component
import ru.ekzeget.mvp.presenters.SplashNewPresenter
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun provideContext(): Context

    fun inject(splashNewPresenter: SplashNewPresenter)
}